### PR TITLE
Pre-release merge - mson-integration on to master branche

### PR DESCRIPTION
**Pull request by klokane**
Big merge of mson-integration on to master branch
Confilict resolved
Ready for review

### CODE REVIEW

File: src/MSONOneOfParser.h [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+#define SNOWCRASH_MSONONEOFPARSER_H
+
+#include "MSONMixinParser.h"
+
+using namespace scpl;
+
+namespace snowcrash {
+
+    /** MSON One Of matching regex */
+    const char* const MSONOneOfRegex = "^[[:blank:]]*[Oo]ne [Oo]f[[:blank:]]*$";
```
Comment (zdne) [inline src/MSONOneOfParser.h]: The white space should tolerate more than just one space (e.g. any white space tabs or an arbitrary number of spaces)

File: src/MSONParameterParser.h [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+        }
+
+        static void parseAttributes(const MarkdownNodeIterator& node,
+                                    SectionParserData& pd,
+                                    const std::vector<mdp::ByteBuffer>& attributes,
+                                    const ParseResultRef<MSONParameter>& out) {
+
+            out.node.use = UndefinedParameterUse;
+
+            if (attributes.size() <= 2) {
```
Comment (zdne) [inline src/MSONParameterParser.h]: @pksunkara why this magic number here (hint: please use constants with an explanatory comment)

File: src/MSONParameterParser.h [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+                        out.node.use = OptionalParameterUse;
+                        definedUse = true;
+                    }
+                    else if (attributes[i] == "required" && !definedUse) {
+                        out.node.use = RequiredParameterUse;
+                        definedUse = true;
+                    }
+                    else {
+                        if (RegexMatch(attributes[i], "^enum\\[[^][]+]$")) {
+                            out.node.type = attributes[i].substr(5, attributes[i].length() - 6);
```
Comment (zdne) [inline src/MSONParameterParser.h]: ditto (magic numbers)

File: test/test-ActionParser.cc [hunk 4]
```diff
@@ -489,3 +489,34 @@ TEST_CASE("Warn when request is not followed by a response", "[action]")
     REQUIRE(action.sourceMap.examples.collection[1].requests.collection.size() == 1);
     REQUIRE(action.sourceMap.examples.collection[1].responses.collection.size() == 0);
 }
+
+// TODO: to be implemented
```
Comment (zdne) [inline test/test-ActionParser.cc]: @pksunkara what is this? This shouldn't get to master.

File: src/MSONOneOfParser.h (hunk 1 above)
Comment (klokane) [inline src/MSONOneOfParser.h]: Fixed, ready for review https://github.com/apiaryio/snowcrash/pull/311

**Comment by klokane** [comment]
By my opinion, ready to merge to master

**Pull request merged**